config/v1/types_ingress.go

```diff
@@ -151,8 +152,28 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// network load balancer. Present only if type is NLB.
+	//
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"nlbParameters,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+	// You can assign Elastic IP addresses to the Network Load Balancer by adding the following annotation.
+	// The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+	// service.beta.kubernetes.io/aws-load-balancer-eip-allocations: eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+	// https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	EIPAllocations []EIPAllocations `json:"eipAllocations"`
+}
+
+type EIPAllocations string
```

Comment:
Calling this the same name as the field type is confusing to me. I don't think this should be plural, it is a singular EIPAllocation.

```suggestion
type EIPAllocation string
```

Reply:
I think EIPAllocation shall be singular as it is array of type EIPAllocation and not EIPAllocation.

Like For example we won't say array of type ints but array of type int.

What do you think ?